vendor/engines/secure_rooms/app/services/secure_rooms/access_manager.rb

```diff
@@ -0,0 +1,48 @@
+module SecureRooms
+
+  class AccessManager
+
+    attr_reader :verdict
+
+    def initialize(user, card_reader, requested_account_id)
+      @user = user
+      @card_reader = card_reader
+      @requested_account_id = requested_account_id
+    end
+
+    def process
+      @verdict = create_verdict
+      event = create_event(verdict)
+    end
+
+    # TODO: Extract
+    def create_verdict
+      SecureRooms::CheckAccess.new.authorize(
```

Comment:
Would it be worth building this in the controller and passing the access checker into the initializer for this as a dependency injection? Not sure if it would help or not.